src/back-end/lib/db/proposal/sprint-with-us.ts

```diff
@@ -2041,4 +2050,59 @@ export const readOneSWUProposalAuthor = tryDb<[Id], User | null>(
   }
 );
 
+/**
+ * This function checks if all proposals for an SWU opportunity are evaluated for the Team Scenario phase
+ * If all proposals are evaluated, it automatically changes the opportunity status to Processing
+ */
+export async function checkAndUpdateSWUOpportunityProcessingStatus(
+  connection: Connection,
+  opportunityId: Id,
+  session: AuthenticatedSession
+): Promise<void> {
+  // Get all active proposals
+  const activeProposals = await generateSWUProposalQuery(connection)
+    .where({ "proposals.opportunity": opportunityId })
+    .whereNotIn("statuses.status", [
+      SWUProposalStatus.Withdrawn,
+      SWUProposalStatus.Disqualified,
+      SWUProposalStatus.Draft,
+      // Exclude proposals that didn't make it to the team scenario phase
+      SWUProposalStatus.Submitted,
+      SWUProposalStatus.UnderReviewTeamQuestions,
+      SWUProposalStatus.DeprecatedEvaluatedTeamQuestions,
+      SWUProposalStatus.UnderReviewCodeChallenge,
+      SWUProposalStatus.EvaluatedCodeChallenge
+    ]);
+
+  // Get current opportunity status
+  const currentOpportunity = await generateSWUOpportunityQuery(connection)
+    .where({ "opportunities.id": opportunityId })
+    .select("statuses.status")
+    .first();
+
+  if (!currentOpportunity) {
+    return; // Opportunity not found
```

Comment:
Could see this being hard to debug if the opportunity isn't found for whatever reason.